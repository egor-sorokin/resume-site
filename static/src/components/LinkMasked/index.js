import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';


const LinkStretched = ({linkData, cssClasses, target}) => (
  <div className="link-masked">
    <span className={`link-masked__placeholder ${cssClasses}`}>{linkData.text}</span>
    <div className="link-masked__mask">
      <a
        href={linkData.url}
        target={target}
        className={`link-masked__text ${cssClasses}`}
      >
        {linkData.text}
      </a>
    </div>
  </div>
);

LinkStretched.propTypes = {
  linkData: PropTypes.shape({}),
  cssClasses: PropTypes.string,
  target: PropTypes.string
};

LinkStretched.defaultProps = {
  linkData: {},
  cssClasses: '',
  target: ''
};

export default LinkStretched;