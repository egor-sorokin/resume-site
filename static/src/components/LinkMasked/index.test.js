// eslint-disable-next-line
import {shallow, render, mount} from 'enzyme';
import React from 'react';
import LinkMasked from './index';


describe('<LinkMasked />', () => {
  let component;
  const linkData = {
    label: 'Label test',
    text: 'Text test',
    url: 'example.url'
  };


  describe('base tests', () => {
    beforeEach(() => {
      component = shallow(<LinkMasked />);
    });

    
    it('renders without crashing', () => {
      expect(component).toBeDefined();
    });


    it('matches snapshot', () => {
      expect(component).toMatchSnapshot();
    });


    it('renders 2 components', () => {
      expect(component.children()).toHaveLength(2);
    });
  });


  describe('traversions tests', () => {
    beforeEach(() => {
      component = shallow(<LinkMasked linkData={linkData}/>);
    });


    it('checks a sameness of a link placeholder and a mask text', () => {
      expect(component.find('.link-masked__placeholder').text()).toEqual(linkData.text);
      expect(component.find('.link-masked__text').text()).toEqual(linkData.text);
    });
  });
});
