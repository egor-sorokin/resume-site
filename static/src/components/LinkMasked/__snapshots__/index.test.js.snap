// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkMasked /> base tests matches snapshot 1`] = `
<div
  className="link-masked"
>
  <span
    className="link-masked__placeholder "
  />
  <div
    className="link-masked__mask"
  >
    <a
      className="link-masked__text "
      href=""
      target=""
    />
  </div>
</div>
`;
